File: client/src/boundary/interactor/GameStatsD2BConverter.ts
import { Converter } from "utils/Converter";
import { GameStats } from "domain/index";
import { BoundaryGameStats } from "boundary/model/index";

export class GameStatsD2BConverter extends Converter<
  GameStats,
  BoundaryGameStats
> {
  convert(gameStats: GameStats): BoundaryGameStats {
    return new BoundaryGameStats(
      gameStats.hitsRemaining,
      gameStats.shipsDestroyed
    );
  }
}

File: server/battleships/use_cases/implementation/src/main/java/eu/optas/use_cases/implementation/GameStatsD2BConverter.java
package eu.optas.use_cases.implementation;

import eu.optas.domain.GameStats;
import eu.optas.use_cases.api.BoundaryGameStats;
import eu.optas.utils.Converter;

public class GameStatsD2BConverter extends Converter<GameStats, BoundaryGameStats> {
    @Override
    public BoundaryGameStats convert(GameStats gameStats) {
        return new BoundaryGameStats(gameStats.getHitsRemaining(), gameStats.getShipsDestroyed());
    }
}
